.editorconfig

```diff
@@ -0,0 +1,9 @@
+root = true
+
+[*]
+indent_style = space
+indent_size = 4
+end_of_line = lf
+charset = utf-8
+insert_final_newline = true
+trim_trailing_whitespace = true
```

Comment:
Trimming the trailing whitespace should not be done for the language files.

Reply:
These settings only apply to files the compiler can format already, it just uses the settings from the .editorconfig file instead of the settings the user has selected. Language files aren't being formatted by the editor, so they won't be affected.

I can only confirm this for Visual Studio though. It's probably best to test this with other popular IDEs first.

Reply:
If it does cause issues with other editors, another `.editorconfig` file could be created in the the language directory to unset this option.

Reply:
You can set preferences by file type, right? We could just not trim for filenames matching `*.txt`

Reply:
That's probably best. @manuelVo Could you add that?

Reply:
Of course, that's much better! `[*.{c,cpp,h,hpp}]` should do the trick.

Reply:
Don't forget `.m` and `.mpp`.

Reply:
I think the .txt exclusion solution you've already added should suffice, really.

Reply:
(In case there are other .txt files, that would need to be trimmed - if I understood correctly)